scripts/prepare.ts

```diff
@@ -91,14 +85,14 @@ for (const network of networks) {
     }
 
     const infoArray = getArtifactInfo(artifact, network);
-    infoArray.forEach(({ version }, i) => {
-      const abiName = i === 0 ? pn : `${pn}-${version}`;
+    infoArray.forEach(({ version }) => {
+      const abiName = version === "0.1.0" ? pn : `${pn}-${version}`;
```

Comment:
If a contract was deployed on a network directly for version `0.2.0` but never for `0.1.0`, then the name would be set without the version, erasing the `0.1.0` ABI version.